web-services/query/src/main/java/datawave/webservice/query/runner/BasicQueryBean.java

```diff
@@ -0,0 +1,388 @@
+                                querySyntax.add(o.toString());
+                        }
+                    } catch (Exception e) {
+                        log.warn("Unable to get query syntax for query logic: " + l.getClass().getCanonicalName());
+                    }
+                    if (querySyntax.isEmpty()) {
+                        querySyntax.add("CUSTOM");
+                    }
+                    d.setQuerySyntax(querySyntax);
+                    
+                    break;
+                    
+                }
+            } catch (Exception e) {
+                log.error("Error setting query logic description", e);
+            }
+        }
+        
+        Principal p = ctx.getCallerPrincipal();
+        String authSting = AuthorizationsUtil.buildUserAuthorizationString(p);
+        response.setAuthString(authSting);
+        response.setTheQueryLogicDescription(theQld);
+        
+        return response;
+    }
+    
+    /**
+     * Display the query plan and link to basic query results for a simple query web UI in the quickstart
+     *
+     * @HTTP 200 Success
+     * @return datawave.webservice.result.QueryWizardStep3Response
+     * @RequestHeader X-ProxiedEntitiesChain use when proxying request for user, by specifying a chain of DNs of the identities to proxy
+     * @RequestHeader X-ProxiedIssuersChain required when using X-ProxiedEntitiesChain, specify one issuer DN per subject DN listed in X-ProxiedEntitiesChain
+     * @ResponseHeader X-OperationTimeInMS time spent on the server performing the operation, does not account for network or result serialization
+     */
+    @POST
+    @Produces({"application/xml", "text/xml", "application/json", "text/yaml", "text/x-yaml", "application/x-yaml", "text/html"})
+    @Path("/{logicName}/showQueryWizardStep3")
+    @GenerateQuerySessionId(cookieBasePath = "/DataWave/BasicQuery/")
+    @Interceptors({ResponseInterceptor.class})
+    @Timed(name = "dw.query.showQueryWizardStep3", absolute = true)
+    public QueryWizardStep3Response showQueryWizardStep3(@Required("logicName") @PathParam("logicName") String logicName,
+                    MultivaluedMap<String,String> queryParameters, @Context HttpHeaders httpHeaders) {
+        CreateQuerySessionIDFilter.QUERY_ID.set(null);
+        GenericResponse<String> createResponse;
+        QueryWizardStep3Response queryWizardStep3Response = new QueryWizardStep3Response();
+        try {
+            createResponse = queryExecutor.createQuery(logicName, queryParameters, httpHeaders);
+        } catch (Exception e) {
+            queryWizardStep3Response.setErrorMessage(e.getMessage());
+            return queryWizardStep3Response;
+        }
+        String queryId = createResponse.getResult();
+        CreateQuerySessionIDFilter.QUERY_ID.set(queryId);
+        queryWizardStep3Response.setQueryId(queryId);
+        
+        if (!logicName.equals("ContentQuery")) {
```

Comment:
Sorry, finally getting back to reviewing this. For the comparison here, it would be better to use your queryLogicFactory instance to get the logicClass, rather than use the logicName. The name is subject to change based on user preference. For example, the user might decide they like the name "ContentLookup" better and reconfigure QLF.xml. But if we rename **datawave.query.tables.content.ContentQueryTable** that'd at least get caught by the compiler here

Second issue: I did try out the update, and I did get past the original issue that I reported, but I then noticed the ContentQuery response fields get truncated to just a few visible chars in the results view. However, the d-column entries in this case are not well-suited for row/col table viewing anyway....because they can be huge, they're base64 encoded, etc. So, I think I'm fine with leaving that problem for another release down the road

Reply:
Thank you @keith-ratcliffe.  I will get these changes made soon.

Reply:
@keith-ratcliffe , I am testing a solution to the logic name issue you mention above.   Basically I get the query logic class from the queryLogicFactory bean.  The problem is that I have to string match the object's class name against "datawave.query.tables.content.ContentQueryTable" which will be detected at runtime, not compile time.  I could do an "instanceof" comparison with ContentQueryTable but that will cause a circular dependency issue with datawave-query-core and datawave-ws-query (by having to import ContentQueryTable in my bean) .  I will push the code for you to review.